SPECS/nodejs/nodejs.spec

```diff
@@ -1,56 +1,91 @@
-Version:        14.17.2
-Release:        2%{?dist}
+Version:        16.14.0
+Release:        1%{?dist}
 License:        BSD and MIT and Public Domain and naist-2003
 Group:          Applications/System
 Vendor:         Microsoft Corporation
 Distribution:   Mariner
 URL:            https://github.com/nodejs/node
+# !!!! Nodejs code has a vendored version of OpenSSL code that must be removed from source tarball 
+# !!!! because it contains patented algorithms.
+# !!!  => use clean-source-tarball.sh script to create a clean and reproducible source tarball.
 Source0:        https://nodejs.org/download/release/v%{version}/node-v%{version}.tar.xz
-Patch0:         patch_tls_nodejs14.patch
-
-BuildRequires:  coreutils >= 8.22, openssl-devel >= 1.0.1
+Patch0:         disable-tlsv1-tlsv1-1.patch
+
+BuildRequires:  brotli-devel
+BuildRequires:  coreutils >= 8.22
+BuildRequires:  gcc
+BuildRequires:  make
+BuildRequires:  ninja-build
+BuildRequires:  openssl-devel >= 1.1.1
 BuildRequires:  python3
 BuildRequires:  which
+BuildRequires:  zlib-devel
+
+Requires:       brotli
 Requires:       coreutils >= 8.22
-Requires:       openssl >= 1.0.1
+Requires:       openssl >= 1.1.1
 Requires:       python3
 
 Provides:       npm = %{npm_version}.%{version}-%{release}
 
 %description
-Node.js is a JavaScript runtime built on Chrome's V8 JavaScript engine. Node.js uses an event-driven, non-blocking I/O model that makes it lightweight and efficient. The Node.js package ecosystem, npm, is the largest ecosystem of open source libraries in the world.
+Node.js is a JavaScript runtime built on Chrome's V8 JavaScript engine.
+Node.js uses an event-driven, non-blocking I/O model that makes it lightweight and efficient.
+The Node.js package ecosystem, npm, is the largest ecosystem of open source libraries in the world.
 
 %package        devel
 Summary:        Development files node
 Group:          System Environment/Base
 Requires:       %{name} = %{version}-%{release}
+Requires:       brotli-devel
+Requires:       openssl-devel >= 1.1.1
+Requires:       zlib-devel
 
 %description    devel
 The nodejs-devel package contains libraries, header files and documentation
 for developing applications that use nodejs.
 
 %prep
-%setup -q -n node-v%{version}
-%patch0 -p1
+%autosetup -p1 -n node-v%{version}
 
 %build
-sh configure --prefix=%{_prefix} \
-           --shared-openssl \
-           --shared-zlib
-
-make %{?_smp_mflags}
+# remove unsupported TLSv1.3 cipher:
+#    Mariner's OpenSSL configuration does not allow for this TLSv1.3
+#    cipher. OpenSSL does not like being asked to use TLSv1.3 ciphers
+#    it doesn't support (despite being fine processing similar cipher
+#    requests for TLS < 1.3). This cipher's presence in the default
+#    cipher list causes failures when initializing secure contexts
+#    in the context of Node's TLS library.
+sed -i '/TLS_CHACHA20_POLY1305_SHA256/d' ./src/node_constants.h
```

Comment:
Suggestion: sor the sake of consistency, I'd suggest we do it as a patch the way we did it in 1.0 (see: #2463). Especially since we're already removing support for other cyphers through a patch.

Reply:
I'd rather stay with this; it will ease version upgrades since we will not have to change the patch. The other one was kind of compulsory because changes are not easily doable using sed